static/app/utils/queryClient.tsx

```diff
@@ -50,10 +72,15 @@ function useQuery<TQueryFnData, TError = RequestError, TData = TQueryFnData>(
       query: endpointOptions?.query,
     });
 
-  const queryFn =
-    typeof queryFnOrQueryOptions === 'function' ? queryFnOrQueryOptions : defaultQueryFn;
+  const queryFn = isQueryFn(queryFnOrQueryOptions)
+    ? queryFnOrQueryOptions
+    : defaultQueryFn;
+
+  const options =
+    queryOptions ??
+    (!isQueryFn(queryFnOrQueryOptions) ? queryFnOrQueryOptions : undefined);
```

Comment:
I'd drop the `!` so there's less logic to invert when reading.

```suggestion
    (isQueryFn(queryFnOrQueryOptions) ? undefined : queryFnOrQueryOptions);
```